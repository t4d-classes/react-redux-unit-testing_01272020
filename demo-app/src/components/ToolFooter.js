import React from 'react';
import PropTypes from 'prop-types';

export const ToolFooter = ({ companyName }) => {

  return <footer>
    <small>&copy; {new Date().getFullYear()} {companyName}. All Rights Reserved.</small>
  </footer>

};

ToolFooter.propTypes = {
  companyName: PropTypes.string.isRequired,
};