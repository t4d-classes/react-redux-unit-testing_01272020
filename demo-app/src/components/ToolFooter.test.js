import * as React from 'react';
import { render, mount } from 'enzyme';

import { ToolFooter } from './ToolFooter';

describe('Tool Footer Component', () => {

  const companyName = 'A Cool Company, Inc.';
  const copyrightYear = new Date().getFullYear();

  beforeEach(() => {
    console.log('Before Each Tool Footer');
  });

  describe('Static Rendering', () => {

    beforeEach(() => {
      console.log('Before Each Tool Footer - Static Rendering');
    });

    test('<ToolFooter /> renders', () => {
      
      const componentHtml = render(
        <div><ToolFooter copyrightYear={copyrightYear} companyName={companyName} /></div>
      ).html();
      
      expect(componentHtml).toMatchSnapshot();
    });

  });

  describe('DOM Rendering', () => {

    beforeEach(() => {
      console.log('Before Each Tool Footer - DOM Rendering');
    });

    test('<ToolFooter /> mounts', () => {
      
      const component = mount(
        <ToolFooter copyrightYear={copyrightYear} companyName={companyName} />
      );

      const componentDOMNode = component.find('small');

      expect(componentDOMNode.text()).toBe(`Â© ${copyrightYear} ${companyName}. All Rights Reserved.`);
    });

  });

});