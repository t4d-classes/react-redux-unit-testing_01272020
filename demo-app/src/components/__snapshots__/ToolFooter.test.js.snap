// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tool Footer Component Static Rendering <ToolFooter /> renders 1`] = `"<footer><small>&#xA9; 2020 A Cool Company, Inc.. All Rights Reserved.</small></footer>"`;
